CRM/Contribute/Form/ContributionPage/Amount.php

```diff
@@ -337,6 +335,7 @@ public static function formRule($fields, $files, $self) {
     if (isset($fields['is_pay_later'])) {
       if (empty($fields['pay_later_text'])) {
         $errors['pay_later_text'] = ts('Please enter the text for the \'pay later\' checkbox displayed on the contribution form.');
+        $this->addElement('checkbox', 'is_monetary', ts('Execute real-time monetary transactions'));
```

Comment:
My fundamental belief is that that checkbox should be deprecated out of the schema. As far as I can tell if $amount > 0 then we should treat it as monetary so I would attempt to calculate that where possible.